pkg/provider/gcp/secretmanager/push_secret.go

```diff
@@ -34,11 +34,12 @@ const (
 	PushSecretMetadataMergePolicyMerge   PushSecretMetadataMergePolicy = "Merge"
 )
 
-type Metadata struct {
-	Annotations map[string]string             `json:"annotations"`
-	Labels      map[string]string             `json:"labels"`
+type PushSecretMetadataSpec struct {
```

Comment:
Oh snap, I'm so sorry, I didn't realise that metadata here was already a thing.

Thank you for the refactor though. This just means that we'll have to release this in a minor version `0.12.0` because it's a breaking change. Existing specs will need to redefine their metadata field.

Reply:
Yes, you're right! I discovered the existing metadata pattern when implementing CMEK support. This makes the code more consistent with the rest of the codebase and follows the established patterns.

I agree this should be a minor version bump to 0.12.0 since users will need to update their PushSecret specs to use the structured metadata format. I'll update the PR description to:
1. Highlight this as a breaking change
2. Provide clear migration examples for users
3. Note the 0.12.0 version requirement

I will add a section in the docs about migrating from the old metadata format to the new structured format.